### PR TITLE
Enable Debian/Sid Build

### PR DESCRIPTION
**Pull request by myungjoo**
Enable Debian/Sid Build.

Build & Unit Tested in Debian/Sid

commit 7b088be511bc0cf821879821f71d9d46989a53d5 (HEAD -> debian/1, github-fork/debian/1)
Author: MyungJoo Ham <myungjoo.ham@samsung.com>
Date:   Wed Jul 7 19:53:39 2021 +0900

    Debian: add debian release changelog for Sid
    
    Add Debian Sid as debian release target.
    Sid is the "devel" version of Debian.
    
    Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>

commit ce73a85b0b0341ffa9174cabb5efc737fac57603
Author: MyungJoo Ham <myungjoo.ham@samsung.com>
Date:   Mon Jul 5 15:55:59 2021 +0900

    Debian: enable armnn in Sid.
    
    Debian-Sid has armnn without pkgconfig.
    Allow to use armnn without pkgconfig file.
    
    Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>

commit 468ad45fb8b6761116d758afa601dc4328901157
Author: MyungJoo Ham <myungjoo.ham@samsung.com>
Date:   Mon Jul 5 14:19:29 2021 +0900

    Meson: use flatbuf if it's >= 2.0.0
    
    Restrict flatbuf version to 2.0.0 or higher.
    Older flatbuf incurs test case errors.
    Disable related test cases, too.
    
    Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>

commit 072d17ab1b01b75794da005973e615f0fdc136c2
Author: MyungJoo Ham <myungjoo.ham@samsung.com>
Date:   Mon Jul 5 12:30:30 2021 +0900

    Ubuntu: Xenial build fail with pdebuild fixed.
    
    Xenial appears to identify build-deps before dh_clean.
    Keep Ubuntu-Xenial control file as default for this issue.
    
    In Ubuntu-Bionic or Debian-Sid, there is no such issues.
    
    Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>

commit 8d431bb7087276b2ee315d6cd900bf035a740aab
Author: MyungJoo Ham <myungjoo.ham@samsung.com>
Date:   Mon Jul 5 12:06:53 2021 +0900

    Debian: disable edge-tpu and openvino.
    
    Debian does not have edge-tpu or openvino ready, yet.
    
    Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>

commit 7542fafb7fbbba8fe45f04a71c3a92efa7e167ac
Author: MyungJoo Ham <myungjoo.ham@samsung.com>
Date:   Mon Jul 5 11:50:09 2021 +0900

    Debian: skip SSAT for Debian
    
    We have not duploaded SSAT for Debian.
    Skip SSAT for Debian.
    
    Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>

commit 1c4580bff6b2fc613fc6286e8b59b5ccf217d055
Author: MyungJoo Ham <myungjoo.ham@samsung.com>
Date:   Thu Jul 1 11:00:51 2021 +0900

    Debian: draft Debian distro build files.
    
    Provided packages are a bit different between Ubuntu and Debian.
    Provide a different control file for Debian builds.
    
    Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>

commit 733fa207ac64f9ec7caf077e23008ea54954c63e
Author: MyungJoo Ham <myungjoo.ham@samsung.com>
Date:   Wed Jun 30 18:35:20 2021 +0900

    Debian: multiple control files for different distros.
    
    Build-deps cannot be dynamically replaced per distro.
    Thus, we need to prepare multiple control files and
    slap in appropriate control files with rules.
    
    This allow to prepare Debian and Ubuntu simultaneously.
    
    Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>

### CODE REVIEW

**Comment by myungjoo** [comment]
Status:

Debian-Sid: build OK. test fail (flexbuf TC)
Ubuntu-Bionic: build OK. test OK

**Comment by myungjoo** [comment]
TODO: cut down subpackages for debian.

**Comment by myungjoo** [comment]
TODO: for older meson bugs, we need
```
--- a/meson.build
+++ b/meson.build
@@ -429,8 +429,8 @@ if not get_option('python3-support').disabled()
     python3_inc_valid_args = []
 
     foreach python3_inc_arg : python3_inc_args
-      if cxx.has_argument(python3_inc_arg) and \
-          cxx.check_header('numpy/arrayobject.h', args : python3_inc_arg, dependencies : python3_dep)
+      if (cxx.has_argument(python3_inc_arg) and \
+          cxx.check_header('numpy/arrayobject.h', args : python3_inc_arg, dependencies : python3_dep))
         python3_inc_valid_args += python3_inc_arg
         have_python3 = true
         break
```

**Comment by myungjoo** [comment]
TODO: for flatbuf-less systems:
```
diff --git a/meson.build b/meson.build
index 5b99dbfa..9181ef79 100644
--- a/meson.build
+++ b/meson.build
@@ -141,6 +141,7 @@ protobuf_dep = dependency('protobuf', version: '>= 3.6.1', required: false)
 # Flatbuffers compiler and libraries
 flatc = find_program('flatc', required : get_option('flatbuf-support'))
 flatbuf_dep = disabler()
+flatbuf_version_check_dep = disabler()
 if flatc.found()
   flatc_ver = run_command(flatc, '--version').stdout().split()[2]
   flatbuf_dep = dependency('flatbuffers', version: flatc_ver,

```

**Comment by myungjoo** [comment]
TODO: revert 0.49 commit. SID uses > 0.50

**Comment by myungjoo** [comment]
TODO:
```
--- a/tests/meson.build
+++ b/tests/meson.build
@@ -164,7 +164,7 @@ if gtest_dep.found()
   if armnn_support_is_available
     unittest_filter_armnn = executable('unittest_filter_armnn',
       join_paths('nnstreamer_filter_armnn', 'unittest_filter_armnn.cc'),
-      dependencies: [glib_dep, gst_dep, nnstreamer_dep, gtest_dep, armnn_plugin_dep],
+      dependencies: [glib_dep, gst_dep, nnstreamer_dep, gtest_dep, armnn_plugin_dep, armnn_dep],
       install: get_option('install-test'),
       install_dir: unittest_install_dir
     )

```

**Comment by myungjoo** [comment]
Build & Unit Tested in Debian-Sid

**Comment by myungjoo** [comment]
ISSUE: unit test passes in the shell. it fails in pdebuild.

File: meson.build [hunk 1]
```diff
@@ -230,6 +233,22 @@ if not get_option('mqtt-support').disabled()
...
+# armnn
+armnn_dep = dependency('', required: false)
+if not get_option('armnn-support').disabled()
+  armnn_dep = dependency('armnn', required: false)
+  if (not armnn_dep.found())
+    armnn_dep = dependency('Armnn', required: false)
+  endif
+  if (not armnn_dep.found())
+    armnn_lib = cxx.find_library('armnn', required: false)
+    pthread_lib = cxx.find_library('pthread', required: false)
```
Comment (jaeyun-jung) [inline meson.build:245]: line 136 thread_dep declared (https://mesonbuild.com/Threads.html)

**Comment by myungjoo** [comment]
Pdebuild for Debian/Sid successful.

File: meson.build (hunk 1 above)
Comment (myungjoo) [reply meson.build]: Resolved. Thanks!

**Review by jaeyun-jung — APPROVED** [review]

**Review by wooksong — APPROVED** [review]
Looks good to me. Great job!

**Pull request merged**